core/src/main/java/org/apache/cloudstack/direct/download/HttpsDirectTemplateDownloader.java

```diff
@@ -183,7 +186,7 @@ public Long getRemoteFileSize(String url, String format) {
                 SSLContext context = getSSLContext();
                 urlConnection.setSSLSocketFactory(context.getSocketFactory());
                 urlConnection.connect();
-                boolean isCompressed = !url.endsWith("qcow2");
+                boolean isCompressed = UriUtils.COMMPRESSION_FORMATS.stream().anyMatch(f -> !url.toLowerCase().endsWith(f));
                 return QCOW2Utils.getVirtualSize(urlObj.openStream(), isCompressed);
```

Comment:
should this be a method in `QCOW2Utils`?
the exact same sequence is executed there